src/run_speedy.jl

```diff
@@ -1,50 +1,47 @@
 """
-    Prog = run_speedy(NF,kwargs...)
+    prog_vars = run_speedy(NF,kwargs...)
 
 Runs SpeedyWeather.jl with number format `NF` and any additional parameters in the keyword arguments
 `kwargs...`. Any unspeficied parameters will use the default values as defined in `src/parameters.jl`."""
-function run_speedy(::Type{NF}=Float64;         # number format, use Float64 as default
-                    kwargs...                   # all additional non-default parameters
+function run_speedy(::Type{NF}=Float64;             # number format, use Float64 as default
+                    kwargs...                       # all additional non-default parameters
                     ) where {NF<:AbstractFloat}
 
-    # initialize the model setup, precalculate constants
-    P = Parameters(NF=NF,kwargs...)
-    C = Constants(P)
-    G = GeoSpectral(P)
-    B = Boundaries(P)
-    HD = HorizontalDiffusion(P,C,G,B)
-    M = Model(P,C,G,B,HD)               # pack all of the above into a Model struct
+    # INITALIZE MODEL
+    prog_vars,diag_vars,model_setup = initialize_model(NF,kwargs...)
 
-    # initialize variables
-    Prog = initial_conditions(P,B,G)
-    # Diag = DiagnosticVariables(P)
-
-    # start speedy model integration
-    # time_stepping!(Prog,Diag,M)
-
-    return Prog
+    # START MODEL INTEGRATION
+    time_stepping!(prog_vars,diag_vars,model_setup) 
+    return prog_vars                                # return prognostic variables when finished
 end
 
-"temporary function, testing out different structure"
-function initialize_model(::Type{NF}=Float64;         # number format, use Float64 as default
-                          kwargs...                   # all additional non-default parameters
+"""
+    prog_vars, diag_vars, model_setup = initialize_model(NF,kwargs...)
+    
+Initialize the model by returning
+- `prog_vars`, the initial conditions of the prognostic variables
+- `diag_vars`, the preallocated the diagnotic variables (initialised to zero)
+- `model_setup`, the collected pre-calculated structs that don't change throughout integration:
+parametes, constants, geometry, spectral transform, boundaries, diffusion."""
+function initialize_model(::Type{NF}=Float64;       # number format, use Float64 as default
```

Comment:
@tomkimpson I like the idea to have a separate `initialize_model` function, which is also helpful for testing. Hence, I've integrated that into `run_speedy`